hal/ia32/exceptions.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * Exception handling
+ *
+ * Copyright 2012, 2016, 2021 Phoenix Systems
+ * Copyright 2001, 2005-2006 Pawel Pisarczyk
+ * Author: Pawel Pisarczyk, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+
+
+/* Number of exceptions */
+#define SIZE_EXCEPTIONS 32
+
+
+typedef struct {
+	unsigned int dr0;
+	unsigned int dr1;
+	unsigned int dr2;
+	unsigned int dr3;
+	unsigned int dr4;
+	unsigned int dr5;
+	unsigned int edi;
+	unsigned int esi;
+	unsigned int ebp;
+	unsigned int edx;
+	unsigned int ecx;
+	unsigned int ebx;
+	unsigned int eax;
+	unsigned short gs;
+	unsigned short fs;
+	unsigned short es;
+	unsigned short ds;
+	unsigned int err;
+	unsigned int eip;
+	unsigned int cs;
+	unsigned int eflags;
+	unsigned int esp;
+	unsigned int ss;
```

Comment:
I think there should be u16, u32 types used

Reply:
Fixed.